scripts/irods/test/test_pam_password.py

```diff
@@ -0,0 +1,340 @@
+            self.fail('OS user "[{}]" with password "[{}]" must exist in order to run these tests.'.format(
+                auth_user, auth_pass))
+
+        self.auth_session = session.mkuser_and_return_session('rodsuser', auth_user, auth_pass, lib.get_hostname())
+        self.service_account_environment_file_path = os.path.join(
+            os.path.expanduser('~'), '.irods', 'irods_environment.json')
+
+        self.server_key_path, self.chain_pem_path, self.dhparams_pem_path = test_configurations.generate_default_ssl_files()
+        self.authentication_scheme = 'pam_password'
+        self.configuration_namespace = 'authentication::pam_password'
+
+    @classmethod
+    def tearDownClass(self):
+        self.auth_session.__exit__()
+
+        for filename in [self.chain_pem_path, self.server_key_path, self.dhparams_pem_path]:
+            if os.path.exists(filename):
+                os.unlink(filename)
+
+        self.admin.assert_icommand(['iadmin', 'rmuser', self.auth_session.username])
+        self.admin.__exit__()
+        with session.make_session_for_existing_admin() as admin_session:
+            admin_session.assert_icommand(['iadmin', 'rmuser', self.admin.username])
+
+    def do_test_invalid_password_time_configurations(self, _option_name):
+        # Stash away the original configuration for later...
+        original_config = self.admin.assert_icommand(
+                ['iadmin', 'get_grid_configuration', self.configuration_namespace, _option_name], 'STDOUT')[1].strip()
+
+        IrodsController().stop()
+
+        auth_session_env_backup = copy.deepcopy(self.auth_session.environment_file_contents)
+        admin_session_env_backup = copy.deepcopy(self.admin.environment_file_contents)
+        try:
+            service_account_environment_file_path = os.path.join(
+                os.path.expanduser('~'), '.irods', 'irods_environment.json')
+            with lib.file_backed_up(service_account_environment_file_path):
+                client_update = test_configurations.make_dict_for_ssl_client_environment(
+                    self.server_key_path, self.chain_pem_path, self.dhparams_pem_path)
+
+                lib.update_json_file_from_dict(service_account_environment_file_path, client_update)
+
+                self.admin.environment_file_contents.update(client_update)
+
+                client_update['irods_authentication_scheme'] = self.authentication_scheme
+                self.auth_session.environment_file_contents.update(client_update)
+
+                with temporary_core_file() as core:
+                    core.add_rule(test_configurations.get_pep_for_ssl(self.plugin_name))
+
+                    IrodsController().start()
+
+                    # Make sure the settings applied correctly...
+                    self.admin.assert_icommand(
+                        ['iadmin', 'get_grid_configuration', self.configuration_namespace, _option_name], 'STDOUT')
```

Comment:
If the settings were not applied correctly, I take it this assertion would fail, right?

Does it make sense to assert a specific value in the output?

Reply:
That's right. In the event that the settings were not applied correctly, an error would occur most likely in the SSL negotiations, resulting in a non-zero return code.

That said,  I have updated these calls to look for the output for these configurations as returned by `get_grid_configuration`.